crates/red_knot/src/module.rs

```diff
@@ -388,12 +389,70 @@ pub fn file_to_module(db: &dyn SemanticDb, file: FileId) -> QueryResult<Option<M
 //////////////////////////////////////////////////////
 
 /// Changes the module search paths to `search_paths`.
-pub fn set_module_search_paths(db: &mut dyn SemanticDb, search_paths: Vec<ModuleSearchPath>) {
+pub fn set_module_search_paths(db: &mut dyn SemanticDb, search_paths: ResolvedSearchPathOrder) {
     let jar: &mut SemanticJar = db.jar_mut();
 
     jar.module_resolver = ModuleResolver::new(search_paths);
 }
 
+const TYPESHED_STDLIB_DIRECTORY: &str = "stdlib";
+
+/// A resolved module resolution order, implementing PEP 561
+/// (with some small, deliberate differences)
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct ResolvedSearchPathOrder(Vec<ModuleSearchPath>);
+
+impl Deref for ResolvedSearchPathOrder {
+    type Target = [ModuleSearchPath];
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// Implementation of the module resolution order from PEP-561.
+///
+/// - `extra_paths` is a list of user-provided paths
+///   that should take first priority in the module resolution.
+///   Examples in other type checkers are mypy's MYPYPATH environment variable,
+///   or pyright's stubPath configuration setting.
+/// - `workspace_root` is the root of the workspace,
+///   used for finding first-party modules
+/// - `site-packages` is the path to the user's `site-packages` directory,
+///   where third-party packages from ``PyPI`` are installed
+/// - `custom_typeshed` is a path to standard-library typeshed stubs.
+///   Currently this has to be a directory that exists on disk.
+///   (TODO: fall back to vendored stubs if no custom directory is provided.)
+impl ResolvedSearchPathOrder {
```

Comment:
I would probably call this `ResolvedSearchPathOrder` because an instance of it doesn't implement an order, it's the ordered search paths. 

An alternative would be that `SearchPathOrder` sorts the paths: Given a list of `ModuleSearchPath`, sort the paths according to PEP-561 (using the `ModuleSearchPathKind)

Reply:
> I would probably call this `ResolvedSearchPathOrder`

Not sure I understand -- that's what it's already called

Reply:
I think I see what you mean, though. I renamed it to `OrderedSearchPaths` in https://github.com/astral-sh/ruff/pull/11767/commits/4f0bcf5aa2dfd6a95533e9ec8d0fa87f100d3bd8